tests/arns.test.mjs

```diff
@@ -1509,4 +1512,237 @@ describe('ArNS', async () => {
   });
 
   // TODO: add several error scenarios
+
+  describe('ArNS Discount Eligible Gateways', () => {
+    const getBalances = async ({ memory }) => {
+      const result = await handle(
+        {
+          Tags: [{ name: 'Action', value: 'Balances' }],
+        },
+        memory,
+      );
+
+      const balances = JSON.parse(result.Messages?.[0]?.Data);
+      return balances;
+    };
+
+    const joinedGateway = 'joined-unique-gateway-'.padEnd(43, '0');
+    const firstEpochTimestamp = 1719900000000;
+    const afterDistributionTimestamp =
+      firstEpochTimestamp + 1000 * 60 * 60 * 24 + 1000 * 60 * 40;
+
+    let arnsDiscountMemory;
+    before(async () => {
+      // create gateways
+      const { memory: join1Memory } = await joinNetwork({
+        address: joinedGateway,
+        quantity: 200_000_000_000,
+      });
+
+      // TODO: Join failing gateway -- Submit an observation with another joined gateway that marks the gateway as failing
+      // const failingGateway = 'failing-unique-gateway-'.padEnd(43, '0');
+      // const { memory: join2Memory } = await joinNetwork(
+      //   {
+      //     address: failingGateway,
+      //   },
+      //   firstTick.Memory,
+      // );
+
+      // TODO: We should be able to create and distribute an epoch within one tick. But for now, this errors -- so we tick twice
```

Comment:
What's the nature of the error?

Reply:
I may have solved this error in another PR where `prevEpoch.distributions` was nil. Will try this sgain

Reply:
still doesnt work. specifically the first tick won't cause any distributions, which results in the gateway not receiving tenure

```   'Ticking epoch: 0\n' +
      'Not updating demand factor\n' +
      'Not distributing rewards for last epoch.\n' +
      'Distributions have not occured for the previous epoch. A new epoch will not be created until those are complete: 0\n' +
      'Ticking epoch: 1\n' +
      'Not updating demand factor\n' +
      'Not distributing rewards for last epoch.\n' +
      'Distributions have not occured for the previous epoch. A new epoch will not be created until those are complete: 0\n' +
      '{"Delegated-Supply":0,"New-Demand-Factors":[1,1],"Cron":false,"Message-Id":"mmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmm","_e":1,"Leaving-Gateways-Count":0,"Staked-Supply":0,"Action":"Tick","Locked-Supply":0,"Current-Epoch-Index":0,"Target-Current-Epoch-Index":1,"From-Formatted":"1111111111111111111111111111111111111111111","Circulating-Supply":0,"Epoch-Index":1,"Joined-Gateways-Count":0,"Total-Token-Supply":50000000000000,"From":"1111111111111111111111111111111111111111111","Withdraw-Supply":0,"Protocol-Balance":50000000000000,"Cast":false,"Last-Ticked-Epoch-Index":-1,"Ticked-Epoch-Indexes":[0],"Timestamp":1719988800000}',```

Reply:
does it work if you use the distribution timestamp to create AND distribute the first epoch?

```
      const futureTickResult = await handle(
        {
          Tags: [{ name: 'Action', value: 'Tick' }],
          Timestamp: afterDistributionTimestamp,
        },
        firstTick.Memory,
      );
```

Reply:
yes, the above comment has the log result with that setup. the gateway does not receive tenure in that case